src/blog/public_vs_private_state.md

```diff
@@ -0,0 +1,77 @@
+
+## Aleo State Storage 
+In order to have privacy native to a chain, Aleo uses a record model for application state storage, which is similar to the UTXO model in Bitcoin. However, Aleo provides developers the option to make application states public should they choose to do so. Public states are stored using the account model as done in Ethereum. 
+
+### Storing Private States via Record
+Records are a fundamental data structure that can contain any arbitrary payload and are used for encoding user assets or application states. A record represents a certain state of a program in the global state, for example, the balance of an account or your identity document. 
+
+
+An Aleo record is serialized in the following format:
+
+| Parameter  |             Type             | Description                                                                                         |
+|------------|:----------------------------:|-----------------------------------------------------------------------------------------------------|
+|     owner    |            address           |                      The address public key of the owner of the program record                      |
+|    data    |    `arbitrary`   | A data payload containing arbitrary application-dependent information                               |
+|    nonce   |             group            |                            The serial number nonce of the program record                            |
+
+Records are tied to programs deployed on Aleo, and only the owner has permissions to alter the state of the record.
+
+Records are private by default and are stored as ciphertext on-chain.
+
+### Storing Public State via Mapping
+
+A public state in Aleo is stored in the form of a mapping, which are key-value pairs. Anyone is able to query the state of the mapping by querying the Aleo blockchain explorer.
+
+### Switching Between Privacy States
+By supporting both private and public storage states, an interesting feature emerges in Aleo where states can be converted from private to public and vice versa.
+
+An example of such a use case would be in a poker game. The state of the shuffled deck after dealing the cards to players should be kept private initially. Proceeding, as the "flop" reveals the top 3 cards, the state of those cards should be revealed to the public.
+
+### View Key
+Aleo has a unique feature known as a view key for each account. The view key allows one to decrypt all transactions of it's account. It is different from the private key in that it does not provide the permission to spend the records.
+
+## Private Inputs and Outputs of Programs
+Apart from stored program states being private, the program function inputs and outputs can also be made private of public. The following shows an example of an Aleo program which adds two input numbers. As shown, the developer has the choice to specify if the input and output of a program should be public or private.
+
+```
+function foo:
```

Comment:
Why did you prefer using a build file as example instead of Leo? Do you think that is clearer?

Reply:
Yes, I wanted to show simply that the output can be public or private as well. I'll switch to a Leo example to be more up-to-date.